### DISCUSSION TITLE
How set argument to be using greedy constructors also for class members

### DISCUSSION THREAD

**Discussion by risogolo** [AutoFixture/AutoFixture#1466] [Q&A]
example test method: public async Task AddChatTest_ShouldSaveChatSuccessfuly([Greedy] Chat chat)

where chat has a property of complex type with its own constructor (multiple constructors, and I need also greedy one) that is set via Chat constructor?

**Comment by GeraldLx** [comment]
It depends on your generic or specific approach. You can customize your fixture to always pick the greedy constructor. In one of our legacy applications I went for this approach, as basically every class has an empty and a parametrized ctor. I wanted AutoFixture to always use the second one.

For a no code approach you normally use the fact that AutoFixture creates the parameters from left to right. So you can use the FrozenAttribute to enforce a specific instance.

public async Task AddChatTest_ShouldSaveChatSuccessfuly([Frozen,Greedy] PropertyType propertyType, [Greedy] Chat chat)

This simple approach has the disadvantage that all instances of PropertyType are now the same. If you have the demand for something else, you must write customizations.

**Reply by risogolo** [reply]
In my specific scenario I have created new custom attribute derived from AutoDataAttribute where I set customizations manually for all objects that Iam passing as arguments to testing methods, so they are using GreedyConstructorQuery
